client/state/difm/assemblers.ts

```diff
@@ -0,0 +1,57 @@
+import 'calypso/state/difm/init';
```

Comment:
What is the motivation to go with an assembler file like this? What sort of future logic do you envision further being added into this file? It strangely reminds me of the utility class pattern.

One thing we can do is probably write a `selector` with dependencies as an optional parameter. So whenever DIFM State does not have a value we default to whatever is available in the dependencies. In the world of Redux this is the job of the selector assembling/slicing/resolving the state tree to whatever you want :)

Let me know what you think!

Reply:
An assembler combines different objects into a [Data Transfer Object (DTO)](https://www.martinfowler.com/eaaCatalog/dataTransferObject.html). Specifically, this assembler will construct a DTO to be used in `cart.extra` from various parameters passed to this function. It will not access the redux state directly (that is the job of the selectors). The assembler will also convert keys to snake_case for the DTO.
 
The selector has only one job - to return a particular slice of the redux state. We cannot add DTO constructing logic to a selector.

The assembler is quite different from a utility class since it has a well-defined purpose. (constructing a DTO). You can find more examples in our code base [here](https://github.com/search?q=repo%3AAutomattic%2Fwp-calypso+filename%3Aassembler.js+language%3AJavaScript+language%3AJavaScript&type=Code)

Reply:
Hmmm ok sounds good as long as its a practice followed in the codebase we have nothing to worry!